src/test/java/org/isf/operation/test/Tests.java

```diff
@@ -119,23 +295,693 @@ public void testIoUpdateOperation() throws Exception {
+				diseaseOut2, diseaseOut3, operation, dischargeType, pregTreatmentType,
+				deliveryType, deliveryResult, false);
+
+		OperationRow operationRow1 = new OperationRow(operation, "prescriber", "opResult", new GregorianCalendar(2021, 1, 1), "remarks", admission, new Opd(),
+				null, 10.0F);
+
+		OperationRow operationRow2 = new OperationRow(1, operation, "prescriber", "opResult", new GregorianCalendar(2021, 1, 1), "remarks", admission,
+				new Opd(), null, 10.0F);
+
+		operationRow1.setId(1);
+		assertThat(operationRow1).isEqualTo(operationRow2);
+	}
+
+	@Test
+	public void testOperationRowEquals() throws Exception {
+		OperationType operationType = testOperationType.setup(false);
+		Operation operation = testOperation.setup(operationType, true);
+		OperationRow operationRow = testOperationRow.setup(operation, false);
+
+		assertThat(operationRow.equals(operationRow)).isTrue();
+		assertThat(operationRow).isNotEqualTo(null);
+		assertThat(operationRow).isNotEqualTo("some string");
+	}
+
+	@Test
+	public void testOperationRowHasCode() throws Exception {
+		OperationType operationType = testOperationType.setup(true);
+		Operation operation = testOperation.setup(operationType, false);
+		OperationRow operationRow = testOperationRow.setup(operation, true);
+
+		int hashCode = operationRow.hashCode();
+		// use computed value
+		assertThat(operationRow.hashCode()).isEqualTo(hashCode);
+	}
+
+	@Test
+	public void testOperationRowToString() throws Exception {
+		OperationType operationType = testOperationType.setup(true);
+		Operation operation = testOperation.setup(operationType, false);
+		operation.setDescription("some new description");
+		OperationRow operationRow = testOperationRow.setup(operation, true);
+
+		Ward ward = testWard.setup(false, false);
+		Patient patient = testPatient.setup(false);
+		AdmissionType admissionType = testAdmissionType.setup(false);
+		DiseaseType diseaseType = testDiseaseType.setup(false);
+		Disease diseaseIn = testDisease.setup(diseaseType, false);
+		Disease diseaseOut1 = testDisease.setup(diseaseType, false);
+		diseaseOut1.setCode("888");
+		Disease diseaseOut2 = testDisease.setup(diseaseType, false);
+		diseaseOut2.setCode("777");
+		Disease diseaseOut3 = testDisease.setup(diseaseType, false);
+		diseaseOut3.setCode("666");
+		DischargeType dischargeType = testDischargeType.setup(false);
+		PregnantTreatmentType pregTreatmentType = testPregnantTreatmentType.setup(false);
+		DeliveryType deliveryType = testDeliveryType.setup(false);
+		DeliveryResultType deliveryResult = testDeliveryResultType.setup(false);
+
+		Admission admission = testAdmission.setup(ward, patient, admissionType, diseaseIn, diseaseOut1,
+				diseaseOut2, diseaseOut3, operation, dischargeType, pregTreatmentType,
+				deliveryType, deliveryResult, false);
+		operationRow.setAdmission(admission);
+
+		assertThat(operationRow).hasToString("some new description TestUserId");
 	}
```

Comment:
just wonder , why this big test just for testing  `OperationRow.toString` method, could it not be implicitly covered in any other test?

or maybe  a simplified version of the test, only loading the compromised values like:

	@Test
	public void testOperationRowToStringSimplified() throws Exception {
		OperationType operationType = testOperationType.setup(true);
		Operation operation = testOperation.setup(operationType, false);
		
		operation.setDescription("aDescription");
		OperationRow operationRow = testOperationRow.setup(operation, true);

		Patient patient = testPatient.setup(false);
		Admission admission = testAdmission.setup(null, patient, null, null, null,
				null, null, operation, null, null,
				null, null, false);
		admission.setUserID("UserID");
		operationRow.setAdmission(admission);

		assertThat(operationRow).hasToString("aDescription UserID");
	}

Reply:
Thanks; used your simplified version.

I was under the impression from other reviewers that a single test should test a single method;  that combining things was frowned upon and thus the reason for this test structure.

Reply:
Great.

I am agree about following single responsibility principle by test, but I don't typically add  tests for coverage of setters, getters, equals, or hash, mainly because all of them typically get tested implicitly. 

toString usually fits in same category, but as the case here you can be expecting something else about the returning string.